mongo/src/main/java/org/axonframework/extensions/mongo/eventhandling/deadletter/DeadLetterEntry.java

```diff
@@ -0,0 +1,534 @@
+    }
+
+    /**
+     * Sets the diagnostics.
+     *
+     * @param diagnostics The new diagnostics.
+     */
+    public void setDiagnostics(@Nonnull MetaData diagnostics, @Nonnull Serializer serializer) {
+        Class<?> serializationTarget = String.class;
+        if (serializer.canSerializeTo(DBObject.class)) {
+            serializationTarget = DBObject.class;
+        }
+        this.serializedDiagnostics = serializer.serialize(diagnostics, serializationTarget).getData();
+    }
+
+    /**
+     * Releases the message for processing by another thread or process.
+     */
+    public void clearProcessingStarted() {
+        this.processingStarted = null;
+    }
+
+    /**
+     * Sets some indexes needed for the dead letter collection
+     *
+     * @param deadLetterCollection the {@link MongoCollection} to create the indexes on
+     */
+    public static void ensureDeadLetterIndexes(@Nonnull MongoCollection<Document> deadLetterCollection) {
+        deadLetterCollection.createIndex(
+                new BasicDBObject(PROCESSING_GROUP_KEY, ORDER_ASC),
+                new IndexOptions().unique(false).name("processingGroupIdentifierIndex"));
+        deadLetterCollection.createIndex(
+                new BasicDBObject(PROCESSING_GROUP_KEY, ORDER_ASC)
+                        .append(SEQUENCE_IDENTIFIER_KEY, ORDER_ASC),
+                new IndexOptions().unique(false).name("processingGroupAndSequenceIdentifierIndex"));
+        deadLetterCollection.createIndex(
+                new BasicDBObject(PROCESSING_GROUP_KEY, ORDER_ASC)
+                        .append(SEQUENCE_IDENTIFIER_KEY, ORDER_ASC)
+                        .append(INDEX_KEY, ORDER_ASC),
+                new IndexOptions().unique(true).name("uniqueDeadLetterEntryIndex"));
+    }
+
+    /**
+     * Filter only on processingGroup
+     *
+     * @param processingGroup the processing group
+     * @return a {@link Bson} filter
+     */
+    public static Bson processingGroupFilter(@Nonnull String processingGroup) {
+        return eq(PROCESSING_GROUP_KEY, processingGroup);
+    }
+
+    /**
+     * Filter on processingGroup and sequenceIdentifier
+     *
+     * @param processingGroup    the processing group
+     * @param sequenceIdentifier the sequence identifier
+     * @return a {@link Bson} filter
+     */
+    public static Bson processingGroupAndSequenceIdentifierFilter(
+            @Nonnull String processingGroup,
+            @Nonnull String sequenceIdentifier
+    ) {
+        return and(eq(PROCESSING_GROUP_KEY, processingGroup),
+                   eq(SEQUENCE_IDENTIFIER_KEY, sequenceIdentifier));
+    }
+
+    /**
+     * Filter on processingGroup, sequenceIdentifier and index
+     *
+     * @param processingGroup    the processing group
+     * @param sequenceIdentifier the sequence identifier
+     * @param index              the index
+     * @return a {@link Bson} filter
+     */
+    public static Bson uniqueFilter(
```

Comment:
I think the name `uniqueFilter` is misnamed. I would prefer the term `findOneFilter` or `singleDlqItemFilter`.
The word unique immediately triggers database unique constraints in my head, which is not what this does

Reply:
Ah, reviewing further, I would suggest these name changes:

`uniqueFilter` -> `bySpecificDeadLetter`
`uniqueNotLockedFilter` -> `bySpecificUnclaimedDeadLetter`
`indexGreaterThanFilter` -> `byProcessingGroupAndIdentifierWithHigherIndex`

It reads pretty nicely:

```
UpdateResult result = mongoTemplate.deadLetterCollection().updateOne(
                bySpecificUnclaimedDeadLetter(
                        processingGroup,
                        deadLetter.sequenceIdentifier(),
                        deadLetter.index(),
                        processingStartedLimit),
                updateProcessingStarted(GenericDeadLetter.clock.instant())
        );
```